File: flutter/fn.giwu/lib/models/bookmark.dart
class BookmarkModel {
  const BookmarkModel({
    required this.id,
    required this.bible,
    required this.book,
    required this.chapter,
    required this.verse,
    required this.text,
  });

  final int id;
  final String bible;
  final int book;
  final int chapter;
  final int verse;
  final String text;

  factory BookmarkModel.fromJson(Map<String, dynamic> json) => BookmarkModel(
        id: json['id'] as int,
        bible: json['bible'] as String,
        book: json['book'] as int,
        chapter: json['chapter'] as int,
        verse: json['verse'] as int,
        text: json['text'] as String,
      );

  bool matches(String bible, int book, int chapter, int verse) =>
      this.bible == bible &&
      this.book == book &&
      this.chapter == chapter &&
      this.verse == verse;
}

File: php/api.giwu/app/Models/Bookmark.php
<?php

declare(strict_types=1);

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Bookmark extends Model
{
    protected $fillable = ['user_id', 'bible', 'book', 'chapter', 'verse', 'text'];

    protected function casts(): array
    {
        return [
            'book' => 'integer',
            'chapter' => 'integer',
            'verse' => 'integer',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}
